crates/ruff_linter/src/rules/ruff/rules/sequence_sorting.rs

```diff
@@ -0,0 +1,878 @@
+pub(super) enum DisplayKind<'a> {
+    Sequence(SequenceKind<'a>),
+    Dict { values: &'a Vec<ast::Expr> },
+}
+
+/// Create a string representing a fixed-up single-line
+/// definition of `__all__` or `__slots__` (etc.),
+/// that can be inserted into the
+/// source code as a `range_replacement` autofix.
+pub(super) fn sort_single_line_elements_sequence(
+    kind: &SequenceKind,
+    elts: &[ast::Expr],
+    elements: &[&str],
+    locator: &Locator,
+    sorting_style: &SortingStyle,
+) -> String {
+    assert_eq!(elts.len(), elements.len());
+    let (opening_paren, closing_paren) = kind.surrounding_parens(locator.contents());
+    assert!(
+        elements.len() >= 2,
+        "A sequence with < 2 elements cannot be unsorted"
+    );
+    let last_item_index = elements.len() - 1;
+    let mut result = String::from(opening_paren);
+
+    let mut element_pairs = elements.iter().zip(elts).collect_vec();
+    element_pairs.sort_by(|(elem1, _), (elem2, _)| sorting_style.compare(elem1, elem2));
+    // We grab the original source-code ranges using `locator.slice()`
+    // rather than using the expression generator, as this approach allows
+    // us to easily preserve stylistic choices in the original source code
+    // such as whether double or single quotes were used.
+    for (i, (_, elt)) in element_pairs.iter().enumerate() {
+        result.push_str(locator.slice(elt));
+        if i < last_item_index {
+            result.push_str(", ");
+        }
+    }
+
+    result.push_str(closing_paren);
+    result
+}
+
+/// An enumeration of the possible conclusions we could come to
+/// regarding the ordering of the elements in a display of string literals
+#[derive(Debug, is_macro::Is)]
+pub(super) enum SortClassification<'a> {
+    /// It's a display of string literals that is already sorted
+    Sorted,
+    /// It's an unsorted display of string literals,
+    /// but we wouldn't be able to autofix it
+    UnsortedButUnfixable,
+    /// It's an unsorted display of string literals,
+    /// and it's possible we could generate a fix for it
+    UnsortedAndMaybeFixable { items: Vec<&'a str> },
+    /// The display contains one or more items that are not string
+    /// literals.
+    NotAListOfStringLiterals,
+}
+
+impl<'a> SortClassification<'a> {
+    pub(super) fn of_elements(elements: &'a [ast::Expr], sorting_style: &SortingStyle) -> Self {
+        let Some((first, rest @ [_, ..])) = elements.split_first() else {
+            return Self::Sorted;
+        };
+        let Some(string_node) = first.as_string_literal_expr() else {
+            return Self::NotAListOfStringLiterals;
+        };
+        let mut this = string_node.value.to_str();
+
+        for expr in rest {
+            let Some(string_node) = expr.as_string_literal_expr() else {
+                return Self::NotAListOfStringLiterals;
+            };
+            let next = string_node.value.to_str();
+            if sorting_style.compare(next, this).is_lt() {
```

Comment:
So this is like: we found something that's out-of-order, so now we return all elements? If so, can you add a comment here to explain the logic.